src/pages/groups/Group.js

```diff
@@ -0,0 +1,282 @@
+
+    return h(Modal, {
+      onDismiss,
+      title: 'Edit Roles',
+      okButton: buttonPrimary({
+        onClick: () => this.submit()
+      }, ['Change Role'])
+    }, [
+      div({ style: { marginBottom: '0.25rem' } }, [
+        'Edit role for ',
+        b([email])
+      ]),
+      h(RadioButton, {
+        text: 'Admin', checked: role === 'admin',
+        onChange: () => this.setState({ role: 'admin' })
+      }),
+      h(RadioButton, {
+        text: 'Member', checked: role === 'member',
+        onChange: () => this.setState({ role: 'member' })
+      }),
+      submitting && spinnerOverlay
+    ])
+  }
+
+  async submit() {
+    const { groupName, user: { email }, onSuccess } = this.props
+    const { role } = this.state
+
+    try {
+      this.setState({ submitting: true })
+      await Rawls.group(groupName).changeMemberRole(email, this.props.user.role, role)
+      onSuccess()
+    } catch (error) {
+      this.setState({ submitting: false })
+      reportError('Error updating user', error)
+    }
+  }
+}
+
+const DeleteUserModal = pure(({ onDismiss, onSubmit, userEmail }) => {
+  return h(Modal, {
+    onDismiss,
+    title: 'Confirm',
+    okButton: buttonPrimary({
+      onClick: onSubmit
+    }, ['Delete User'])
+  }, [
+    'Are you sure you want to delete the user ',
+    b(`${userEmail}?`)
+  ])
+})
+
+const MemberCard = pure(({ member: { email, role }, onEdit, onDelete }) => {
+  return div({
+    style: styles.longCard
+  }, [
+    div({ style: { flex: '1' } }, [email]),
+    div({ style: { flex: '0 0 150px', textTransform: 'capitalize' } }, [role]),
+    div({ style: { flex: '0 0 auto', textAlign: 'right' } }, [
+      link({ onClick: onEdit }, ['Edit Role']),
+      ' | ',
+      link({ onClick: onDelete }, ['Remove'])
+    ])
+  ])
+})
+
+const NewUserCard = pure(({ onClick }) => {
+  return h(Clickable, {
+    style: styles.shortCreateCard,
+    onClick
+  }, [
+    div(['Add a User']),
+    icon('plus-circle', { style: { marginTop: '0.5rem' }, size: 21 })
+  ])
+})
+
+export class GroupDetails extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      filter: '',
+      members: null,
+      creatingNewUser: false,
+      editingUser: false,
+      deletingUser: false,
+      updating: false,
+      ...StateHistory.get()
+    }
+  }
+
+  async refresh() {
+    const { groupName } = this.props
+
+    try {
+      this.setState({ isDataLoaded: false, creatingNewUser: false, editingUser: false, deletingUser: false, updating: false })
+      const { membersEmails, adminsEmails } = await Rawls.group(groupName).listMembers()
+      this.setState({
+        isDataLoaded: true,
+        members: _.sortBy('email', _.concat(
+          _.map(adm => ({ email: adm, role: 'admin' }), adminsEmails),
+          _.map(mem => ({ email: mem, role: 'member' }), membersEmails)
+        ))
+      })
+    } catch (error) {
+      reportError('Error loading group list', error)
```

Comment:
TOL: If this fails, we'll end up with an infinite spinner. Not the end of the world, but we should try to avoid that when possible. I suspect that variable names play a role in whether we tend to catch these cases. In this example, the right thing to do would be to set `isDataLoaded: true`, which _sounds_ wrong because the data didn't load successfully. If we choose a different name, e.g. `loading` or `busy`, it might be a bit easier on the brain and help us notice these cases.